### PR TITLE
ScopeConfigInterface can be more than a string

### LINKED ISSUE #29345: [Issue] ScopeConfigInterface can be more than a string
This issue is automatically created based on existing pull request: magento/magento2#29305: ScopeConfigInterface can be more than a string

---------
### Summary (*)
Magento 2.4-develop
Reverts parts of https://github.com/magento/magento2/commit/a5a98dfa56cac8cdb60e15568c10578f522be3f3 as it restricts `@magentoConfigFixture` unnecessarily to strings.

Neither `Magento\Framework\App\Config\ScopeConfigInterface` nor `Magento\Framework\App\Config` have this limitation.

```
    /**
     * Retrieve config value by path and scope
     *
     * @param string $path
     * @param string $scope
     * @param null|int|string $scopeCode
     * @return mixed
     */
    public function getValue(
        $path = null,
        $scope = ScopeConfigInterface::SCOPE_TYPE_DEFAULT,
        $scopeCode = null
    ) {
```

### Examples (*)
1. Check Magento\Config\Model\Config\Backend\Serialized
2. Note the possible return values

### PR DESCRIPTION
**Pull request by ghost**
### Description (*)
Reverts parts of https://github.com/magento/magento2/commit/a5a98dfa56cac8cdb60e15568c10578f522be3f3 as it restricts `@magentoConfigFixture` unnecessarily to strings.

Neither `Magento\Framework\App\Config\ScopeConfigInterface` nor `Magento\Framework\App\Config` have this limitation.

```
    /**
     * Retrieve config value by path and scope
     *
     * @param string $path
     * @param string $scope
     * @param null|int|string $scopeCode
     * @return mixed
     */
    public function getValue(
        $path = null,
        $scope = ScopeConfigInterface::SCOPE_TYPE_DEFAULT,
        $scopeCode = null
    ) {
```

### Fixed Issues (if relevant)
1. N/A

### Manual testing scenarios (*)
1. Check Magento\Config\Model\Config\Backend\Serialized
2. Note the possible return values

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#29345: ScopeConfigInterface can be more than a string

### CODE REVIEW

**Comment by fooman** [comment]
@magento run all tests

File: dev/tests/integration/framework/Magento/TestFramework/Annotation/ConfigFixture.php [hunk 1]
```diff
@@ -71,7 +71,7 @@ protected function _getConfigValue($configPath, $scopeCode = null)
      * @param string|null $scopeCode
      * @return string|null
```
Comment (ihor-sviziev) [inline dev/tests/integration/framework/Magento/TestFramework/Annotation/ConfigFixture.php]:
```suggestion
     * @return mixed|null
```

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-7906](https://jira.corp.magento.com/browse/ENGCOM-7906) has been created to process this Pull Request

**Comment by engcom-Delta** [comment]
**Notice:** QA not applicable
Integration tests are green

**Comment by engcom-Delta** [comment]
@magento create issue

**Pull request merged**